oxenmq/connections.cpp

```diff
@@ -178,7 +180,7 @@ OxenMQ::proxy_connect_sn(std::string_view remote, std::string_view connect_hint,
     p.activity();
     connections_updated = true;
     outgoing_sn_conns.emplace_hint(outgoing_sn_conns.end(), p.conn_id, ConnectionID{remote});
-    auto it = connections.emplace_hint(connections.end(), p.conn_id, std::move(socket));
+    auto it = connections.emplace_hint(connections.end(), p.conn_id, *std::move(socket));
```

Comment:
what about `std::move(*socket)`?

Reply:
It does the same thing (see the overloads at https://en.cppreference.com/w/cpp/utility/optional/operator*)